src/python/pants/util/objects.py

```diff
@@ -333,22 +337,27 @@ def iterate_enum_variants(cls):
       """
       return cls._singletons.values()
 
+    @classmethod
+    def register_option(cls, register, *args, **kwargs):
```

Comment:
Can any of the args be specified explicitly, like `idempotent_type_check`? `*args, **kwargs` params are quite difficult to understand what is expected and what is not valid. It seems we may need to keep it, but even if we could specify one or two more arguments explicitly that would be a win.

Reply:
Do you mean requiring the caller to explicitly specify those arguments when registering an enum option? After some thought that is I think what I arrived at in https://github.com/pantsbuild/pants/pull/7304/files/c4ca5d5134c6d15411b1f67b6c202fe77a977d45#r261845170, which would remove this method altogether for the reasons you've described.

Reply:
No, what I meant is in this function definition making it explicit what the parameters are. The callers can still avoid having to specify every argument by giving default arguments to the function definition.

For example
```python
def register_option(cls, register, is_idempotent, *args, **kwargs):
  pass
```

Is better and more explicit than 
```python
def register_option(cls, register, *args, **kwargs):
  pass
```

Sounds like this is resolved either way.

Reply:
Oh -- I get that now. This bullet was dodged but will keep this in mind for the future.